src/main/java/org/folio/edge/sip2/BootstrapVerticle.java

```diff
@@ -0,0 +1,58 @@
+package org.folio.edge.sip2;
+
+import io.vertx.config.ConfigRetriever;
+import io.vertx.config.ConfigRetrieverOptions;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonObject;
+import java.util.Objects;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class BootstrapVerticle extends AbstractVerticle {
```

Comment:
No, not really.  Making the Bootstrap a verticle allows us to make use of the launcher command line processing. And we need a hook point to wire up the config retriever and fetch the tenant config from s3 before the MainVerticle can be deployed

Reply:
I'm still not sure I understand why this is necessary. Why does the MainVerticle have to be launched after the config data is retrieved?

Since the code is reactive, we could just start the TCP listener when the config has been retrieved in the MainVerticle and the properties merged into the Verticle's config.

When the config changes via the change listener, I think we could update the live config without undeploying/redeploying. I guess the tricky part would be shutting down connections that are no longer part of the config and we should be able to do that since each connection will already know its client IP and each time a message is received we can check against the current CIDR range and kill any connection that is now not present. Something like that.

Undeploying/redploying makes me nervous. We should avoid killing connections when the config changes are unrelated to currently running tenants, even if it is rare.

Reply:
OK, I found a way to do this in one verticle. 
It became a lot easier now that the port & okapiUrl config are back in the bootstrap config and separate from the tenant config.

Reply:
Multi tenant is now done in one verticle